### PR TITLE
feat(web-analytics): Display values from previous period alongside current data

### PR DESCRIPTION
**Pull request by rafaeelaudibert**
## Problem

We want to be able to see at a glance if the current numbers are going up/down compared to the previous period. This came up in a user interview we did yesterday, and we thought it was an easy win.

## Changes

Queries are slightly heavier now, because we need to query for double the period we were doing previously and then split then using `*If` functions.

In the UI, we're simply representing this with a trend icon which displays more information on hover.\\
image
image

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Unit tests for the backend, manual testing for the UI

### CODE REVIEW

**Comment by rafaeelaudibert** [comment]
Don't you think the UI looks too noisy if this is turned on all the time?

File: posthog/hogql_queries/web_analytics/stats_table.py [hunk 1]
```diff
@@ -340,15 +397,17 @@ def to_path_bounce_query(self) -> ast.SelectQuery:
 LEFT JOIN (
     SELECT
         breakdown_value,
-        avg(is_bounce) AS bounce_rate
+        coalesce(avgIf(is_bounce, current_period_segment), 0) AS bounce_rate,
```
Comment (robbie-c) [inline posthog/hogql_queries/web_analytics/stats_table.py:400]: Do we need the coalesce to zero? IMO returning NULL is more accurate

Comment (rafaeelaudibert) [reply posthog/hogql_queries/web_analytics/stats_table.py]: I've changed this back to returning `NULL`

**Comment by robbie-c** [comment]
Maybe icon size
Maybe icon padding
Padding when no arrow or always add arrow
Infinity percent
Tooltip triangle position

**Comment by rafaeelaudibert** [comment]
@robbie-c Ready to rereview, worked on most of the changes you mentioned above

- Icon padding
image


- Infinity percent + tooltip arrow in the right place
image

File: frontend/src/lib/utils.tsx [hunk 2]
```diff
@@ -124,6 +124,10 @@ export function percentage(
     maximumFractionDigits: number = 2,
     fixedPrecision: boolean = false
 ): string {
+    if (division === Infinity) {
```
Comment (robbie-c) [inline frontend/src/lib/utils.tsx:127]:
```suggestion
    if (!isFinite(division)) {
```

Comment (robbie-c) [reply frontend/src/lib/utils.tsx:127]: (js has positive and negative infinity)

**Review by robbie-c — APPROVED** [review]

**Pull request merged**